### PR TITLE
feat: Adding support for date_nanos to Anomaly Detection

### PR DESCRIPTION
**Pull request by jehuty0shift**
### Description
This PR add support for **date_nanos** type for time field in Anomaly Detection. 

### Issues Resolved
[#1226](https://github.com/opensearch-project/anomaly-detection/issues/1226)

 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by jehuty0shift** [comment]
Hi, I'm new to contributing to this repository. I'm open to adding test to more meaningful places and/or to backport this feature to 2.x branch if needed. 

I will also PR the dashboard plugin when this PR get approved (knowing this PR surely needs more work done).

**Comment by kaituo** [comment]
@jehuty0shift Thanks a lot. Will review.

File: src/test/java/org/opensearch/ad/e2e/RuleModelPerfIT.java [hunk 1]
```diff
@@ -50,6 +50,21 @@ public void testRule() throws Exception {
...
+        // TODO: this test case will run for a much longer time and timeout with security enabled
+        if (!isHttps()) {
+            disableResourceNotFoundFaultTolerence();
+            // there are 8 entities in the data set. Each one needs 1500 rows as training data.
+            Map<String, Double> minPrecision = new HashMap<>();
+            minPrecision.put("Phoenix", 0.5);
+            minPrecision.put("Scottsdale", 0.5);
+            Map<String, Double> minRecall = new HashMap<>();
+            minRecall.put("Phoenix", 0.9);
+            minRecall.put("Scottsdale", 0.9);
```
Comment (kaituo) [inline src/test/java/org/opensearch/ad/e2e/RuleModelPerfIT.java:63]: I think it is changed to 0.6 in the main branch. Can you rebase?

Comment (jehuty0shift) [reply src/test/java/org/opensearch/ad/e2e/RuleModelPerfIT.java]: Hello, sure i have rebased and force-pushed the changes.

**Comment by kaituo** [comment]
@ jehuty0shift Once the CI passed, will approve and merge your PR. Please also work on a change on AD dashboard plugin side. I assume the date_nanos timestamp may not show in create/edit detector page.

**Review by kaituo — APPROVED** [review]

**Comment by jehuty0shift** [comment]
Yes, the PR to Anomaly Detection dashboard plugin will be done today. Thank you very much for your review ! Looking forward to contribute more to OpenSearch.

**Pull request merged**